From a4b6130ea15c88ce29876e7455b8901da5a823f7 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@caviumnetworks.com>
Date: Fri, 14 Sep 2018 10:57:36 -0700
Subject: [PATCH 0824/1239] octeontx/tx2: discard usage of soft reset register

Remove usage of soft reset in RST registers instead use
psci calls.

Signed-off-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 arch/arm/include/asm/arch-octeontx/octeontx.h |  2 -
 arch/arm/mach-octeontx/cpu.c                  |  5 +++
 arch/arm/mach-octeontx2/cpu.c                 |  5 +++
 board/cavium/octeontx/octeontx.c              | 10 -----
 board/cavium/octeontx2/octeontx2.c            | 10 -----
 configs/octeontx2_95xx_defconfig              |  1 -
 configs/octeontx2_96xx_defconfig              |  1 -
 configs/octeontx_81xx_defconfig               |  1 -
 configs/octeontx_83xx_defconfig               |  1 -
 drivers/i2c/octeontx_i2c.c                    | 37 +------------------
 10 files changed, 12 insertions(+), 61 deletions(-)

diff --git a/arch/arm/include/asm/arch-octeontx/octeontx.h b/arch/arm/include/asm/arch-octeontx/octeontx.h
index 64bcad14f7..c1e5d35870 100644
--- a/arch/arm/include/asm/arch-octeontx/octeontx.h
+++ b/arch/arm/include/asm/arch-octeontx/octeontx.h
@@ -13,8 +13,6 @@
 #define MAX_LMAC_PER_BGX 4
 
 /** Reg offsets */
-#define CAVM_RST_BOOT		0x87E006001600ULL
-#define CAVM_RST_SOFT_RST	0x87E006001680ULL
 #define CAVM_MIO_FUS_DAT2	0x87E003001410ULL
 
 #define CAVM_GTI_CWD_POKEX	0x844000050000ULL
diff --git a/arch/arm/mach-octeontx/cpu.c b/arch/arm/mach-octeontx/cpu.c
index 8c7cf6a92b..0f544ca716 100644
--- a/arch/arm/mach-octeontx/cpu.c
+++ b/arch/arm/mach-octeontx/cpu.c
@@ -47,3 +47,8 @@ u64 get_page_table_size(void)
 {
 	return 0x4c000;
 }
+
+void reset_cpu(ulong addr)
+{
+
+}
diff --git a/arch/arm/mach-octeontx2/cpu.c b/arch/arm/mach-octeontx2/cpu.c
index 5a9f0cbed7..b43a0dfa6d 100644
--- a/arch/arm/mach-octeontx2/cpu.c
+++ b/arch/arm/mach-octeontx2/cpu.c
@@ -46,3 +46,8 @@ u64 get_page_table_size(void)
 {
 	return 0x4c000;
 }
+
+void reset_cpu(ulong addr)
+{
+
+}
diff --git a/board/cavium/octeontx/octeontx.c b/board/cavium/octeontx/octeontx.c
index 5d4110a845..1ffe084745 100644
--- a/board/cavium/octeontx/octeontx.c
+++ b/board/cavium/octeontx/octeontx.c
@@ -183,16 +183,6 @@ int dram_init(void)
 	return 0;
 }
 
-/**
- * Board specific reset that is system reset.
- */
-void reset_cpu(ulong addr)
-{
-	u64 val = readq(CAVM_RST_SOFT_RST) | 1;
-
-	writeq(val, CAVM_RST_SOFT_RST);
-}
-
 /**
  * Board late initialization routine.
  */
diff --git a/board/cavium/octeontx2/octeontx2.c b/board/cavium/octeontx2/octeontx2.c
index 291bf5bff1..d96b874f87 100644
--- a/board/cavium/octeontx2/octeontx2.c
+++ b/board/cavium/octeontx2/octeontx2.c
@@ -133,16 +133,6 @@ int dram_init(void)
 	return 0;
 }
 
-/**
- * Board specific reset that is system reset.
- */
-void reset_cpu(ulong addr)
-{
-	u64 val = readq(CAVM_RST_CHIP_DOM_W1S) | 1;
-
-	writeq(val, CAVM_RST_CHIP_DOM_W1S);
-}
-
 /**
  * Board misc devices initialization routine.
  */
diff --git a/configs/octeontx2_95xx_defconfig b/configs/octeontx2_95xx_defconfig
index b39cfa6826..51841403db 100644
--- a/configs/octeontx2_95xx_defconfig
+++ b/configs/octeontx2_95xx_defconfig
@@ -3,7 +3,6 @@ CONFIG_ARM=y
 CONFIG_ARCH_OCTEONTX2=y
 CONFIG_SYS_TEXT_BASE=0x04000000
 CONFIG_IDENT_STRING=" for Cavium OcteonTX2 CNF95XX ARM V8 Core"
-# CONFIG_PSCI_RESET is not set
 CONFIG_TARGET_OCTEONTX2_95XX=y
 CONFIG_DEBUG_UART=y
 # CONFIG_AHCI is not set
diff --git a/configs/octeontx2_96xx_defconfig b/configs/octeontx2_96xx_defconfig
index a6eedb2e13..eb230d5a4f 100644
--- a/configs/octeontx2_96xx_defconfig
+++ b/configs/octeontx2_96xx_defconfig
@@ -2,7 +2,6 @@ CONFIG_ARM=y
 # CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_OCTEONTX2=y
 CONFIG_SYS_TEXT_BASE=0x04000000
-# CONFIG_PSCI_RESET is not set
 CONFIG_TARGET_OCTEONTX2_96XX=y
 CONFIG_DEBUG_UART=y
 CONFIG_AHCI=y
diff --git a/configs/octeontx_81xx_defconfig b/configs/octeontx_81xx_defconfig
index b8d5c0c97b..4e82de11fd 100644
--- a/configs/octeontx_81xx_defconfig
+++ b/configs/octeontx_81xx_defconfig
@@ -2,7 +2,6 @@ CONFIG_ARM=y
 # CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_OCTEONTX=y
 CONFIG_SYS_TEXT_BASE=0x500000
-# CONFIG_PSCI_RESET is not set
 CONFIG_TARGET_OCTEONTX_81XX=y
 CONFIG_DEBUG_UART=y
 CONFIG_AHCI=y
diff --git a/configs/octeontx_83xx_defconfig b/configs/octeontx_83xx_defconfig
index 1690909f00..7a6edb8678 100644
--- a/configs/octeontx_83xx_defconfig
+++ b/configs/octeontx_83xx_defconfig
@@ -2,7 +2,6 @@ CONFIG_ARM=y
 # CONFIG_ARM64_SUPPORT_AARCH32 is not set
 CONFIG_ARCH_OCTEONTX=y
 CONFIG_SYS_TEXT_BASE=0x2800000
-# CONFIG_PSCI_RESET is not set
 CONFIG_TARGET_OCTEONTX_83XX=y
 CONFIG_DEBUG_UART=y
 CONFIG_AHCI=y
diff --git a/drivers/i2c/octeontx_i2c.c b/drivers/i2c/octeontx_i2c.c
index b5a9511858..1de3ef04bc 100644
--- a/drivers/i2c/octeontx_i2c.c
+++ b/drivers/i2c/octeontx_i2c.c
@@ -71,8 +71,6 @@
 # define CONFIG_SYS_I2C_OCTEONTX_SPEED_11	CONFIG_SYS_I2C_OCTEONTX_SPEED_10
 #endif
 
-#define RST_BOOT	0x87e006001600ll
-
 #define TWSI_THP		24
 
 #define TWSI_SW_TWSI		0x1000
@@ -80,32 +78,6 @@
 #define TWSI_INT		0x1010
 #define TWSI_SW_TWSI_EXT	0x1018
 
-union rst_boot {
-	u64 u;
-	struct {
-		u64 rboot_pin:1;
-		u64 rboot:1;
-		u64 lboot:10;
-		u64 lboot_ext23:6;
-		u64 lboot_ext45:6;
-		u64 reserved_24_29:6;
-		u64 lboot_oci:3;
-		u64 pnr_mul:6;
-		u64 reserved_39_39:1;
-		u64 c_mul:7;
-		u64 reserved_47_54:8;
-		u64 dis_scan:1;
-		u64 dis_huk:1;
-		u64 vrm_err:1;
-		u64 jt_tstmode:1;
-		u64 ckill_ppdis:1;
-		u64 trusted_mode:1;
-		u64 ejtagdis:1;
-		u64 jtcsrdis:1;
-		u64 chipkill:1;
-	} s;
-};
-
 union twsx_sw_twsi {
 	u64 u;
 	struct {
@@ -805,12 +777,10 @@ static int twsi_init(void *baseaddr, unsigned int speed, int slaveaddr)
 	int n_div;
 	int m_div;
 	union twsx_sw_twsi sw_twsi;
-	union rst_boot rst_boot;
 
 	debug("%s(%p, %u, 0x%x)\n", __func__, baseaddr, speed, slaveaddr);
-	rst_boot.u = readq(RST_BOOT);
 
-	io_clock_hz = rst_boot.s.pnr_mul * PLL_REF_CLK;
+	io_clock_hz = octeontx_get_io_clock();
 
 	/* Set the TWSI clock to a conservative TWSI_BUS_FREQ.  Compute the
 	 * clocks M divider based on the SCLK.
@@ -904,12 +874,9 @@ static int octeontx_i2c_set_bus_speed(struct udevice *bus, unsigned int speed)
 	int m_div, n_div;
 	unsigned io_clock_hz;
 	union twsx_sw_twsi sw_twsi;
-	union rst_boot rst_boot;
 	void *baseaddr = twsi->baseaddr;
 
-	rst_boot.u = readq(RST_BOOT);
-
-	io_clock_hz = rst_boot.s.pnr_mul * PLL_REF_CLK;
+	io_clock_hz = octeontx_get_io_clock();
 	debug("%s(%p, %u) io clock: %u\n", __func__, bus, speed, io_clock_hz);
 
 	/* Set the TWSI clock to a conservative TWSI_BUS_FREQ.  Compute the
-- 
2.29.0

